### PR TITLE
Backport Check for 2 different Xcode locations (#9861) to release/5.0

### PR DESCRIPTION
**Pull request by jonfortescue**
Backport 14da5af378931c944abaf3318315f88e7b710489 to release/5.0.

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Review by garath — APPROVED** [review]

File: src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-runner.apple.sh [hunk 1]
```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
```
Comment (MattGal) [inline src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-runner.apple.sh:1]: I don't think just copying a file into this repo is going to solve that issue, there seems to be a lot more history

Release 5.0 has ios-helix-job-payload.sh [which gets renamed to xharness-runner.ios.sh](https://github.com/dotnet/arcade/commits/29ae7a00fff56255b343fdce87811ec7dbe8d5bf/src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-runner.ios.sh?browsing_rename_history=true&new_path=src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-runner.apple.sh&original_branch=main) and later this file name.  Without the SDK changes to include and use this script I don't think this solves your problem.

**Review by MattGal — CHANGES_REQUESTED** [review]
I think this either needs to take in subsequent Xharness changes or be abandoned.

**Comment by jonfortescue** [comment]
the failing build does indeed indicate that it does not. Okay, I'll dive more deeply into this; was hoping just cherry-picking that commit would be enough.

**Comment by premun** [comment]
The way I was backporting the XHarness was copying the whole `xharness-runner` directory and then 2 XHarness MSBuild tasks from Helix SDK. I can do this for release 5.0 if we need this

**Comment by jonfortescue** [comment]
@premun I would appreciate that!

**Pull request closed without merging**